Meter.c

```diff
@@ -78,6 +83,21 @@ int Meter_humanUnit(char* buffer, unsigned long int value, size_t size) {
    return snprintf(buffer, size, "%.*f%c", precision, (double) value / powi, *prefix);
 }
 
+static void GraphMeterMode_printScale(int scaleExp) {
+   if (scaleExp > 86) { // > 99 yotta
+      addstr("inf");
+   } else if (scaleExp < 10) { // <= 512
```

Comment:
`scaleExp < 0` is reserved for now. Of course it's possible to print something like "1/2", "1/4", or "1/8", but the current algorithm never sets `scaleExp` to less than zero. I can't say taking an unsigned argument for `scaleExp` is a good idea.

Reply:
I guessed that you had some reserved meaning in mind, which makes it IMHO even more advisable to handle that case and throw an error, until it is finally used for something else …

Something like:
```c
if (scaleExp < 0) {
   asser(0 && "scaleExp reached reserved value <0. This should not happen.");
} else if …
```

Reply:
Do you suggest me to add two lines like this?
```
if (scaleExp < 0)
   return;
```

Reply:
Any objections on this solution?

```diff
diff --git a/Meter.c b/Meter.c
index 0310a87a..3479dc17 100644
--- a/Meter.c
+++ b/Meter.c
@@ -84,6 +84,11 @@ int Meter_humanUnit(char* buffer, unsigned long int value, size_t size) {
 }
 
 static void GraphMeterMode_printScale(int scaleExp) {
+   if (scaleExp < 0) {
+      asser(0 && "scaleExp has reserved value <0. This should not happen.");
+      scaleExp = 0;
+   }
+
    if (scaleExp > 86) { // > 99 yotta
       addstr("inf");
    } else if (scaleExp < 10) { // <= 512
```

Just resolves the problematic state on normal program flow, but in debug mode gives you a clear diagnostic with stack trace for debugging.

Reply:
@BenBE I'm not a fan of embedding debug message in assert statements, and I prefer a simple return rather than setting `scaleExp = 0` (which will print "1").

Reply:
My way of doing it:

```c
static void GraphMeterMode_printScale(int scaleExp) {
   if (scaleExp > 86) { // > 99 yotta
      addstr("inf");
   } else if (scaleExp < 10) { // <= 512
      if (scaleExp < 0) {
         // Should never happen, but handle it and do nothing just in case.
         assert(scaleExp >= 0);
         return;
      }
      printw("%3u", 1U << scaleExp);
   } else if (scaleExp % 10 <= 6) { // {1|2|4|8|16|32|64}{K|M|G|T|P|E|Z|Y}
      printw("%2u%c", 1U << (scaleExp % 10), Meter_prefixes[(scaleExp / 10) - 1]);
   } else {
      // To keep the 3 character limit, we express in the "fraction" format
      // such as "M/8" (=128K), "G/4" (=256M) and "T/2" (=512G).
      printw("%c/%u", Meter_prefixes[scaleExp / 10], 1U << (10 - (scaleExp % 10)));
   }
}
```

Reply:
That `if` condition contradicts the `assert` and thus the assert would never hit. Please stick with the other version …

Reply:
No, the assertion would always hit inside the `if`-block. The `assert` conditional has to be opposite to the `if`-conditional.

Reply:
How about combining @Explorer09's code with @BenBE's more precise comment?
(The assert is correct that way for those reading up in a few months, years, ...)
```
if (scaleExp < 0) {
         // scaleExp has reserved values <0. This should not happen.
         assert(scaleExp >= 0);
         return;
      }
```

Reply:
@fasterit I thought people more likely want to know why we need to bother handle the case even though an assert statement already covers it.
If I have to handle `scaleExp < 0`, I would rather remove the assert (but keep the comment anyway).

Reply:
@Explorer09 Works fine for me. /DLange